pandas/tools/plotting.py

```diff
@@ -2872,12 +2875,14 @@ def grouped_hist(data, column=None, by=None, ax=None, bins=50, figsize=None,
     -------
     axes: collection of Matplotlib Axes
     """
-    def plot_group(group, ax):
-        ax.hist(group.dropna().values, bins=bins, **kwargs)
+    def plot_group(group, ax, weights=None):
+        if weights is not None:
+            weights=weights.dropna().values
```

Comment:
I suspect this will fail if the missing values in the `weights` column don't align perfectly with the missing values in the `group` column. It might be cleaner to refactor this to drop rows missing either `group` or `weight` earlier on, so that plot_group only has to deal with valid observations.